### PR TITLE
[WIP] Remove InPlace from mutable Set and Map. 

### PR DESCRIPTION
**Pull request by abuggin**
Start addressing scala/collection-strawman/issues/473

This PR is still failing but wanted some input from others.

### CODE REVIEW

**Comment by abuggin** [comment]
One of the problem is 

> scala/src/library/scala/collection/mutable/HashMap.scala:22: class HashMap needs to be abstract, since method mapInPlace in trait IterableOps of type (f: ((K, V)) => (K, V))HashMap.this.type is not defined

`IterableOps` in `collection/mutable/Iterable.scala` defines `mapInPlace` and is extended by lots of classes that implements Iterable[A]. How can we differentiate which collection are supposed to get the `IterableOps` with `*InPlace` and  which one are not?

**Comment by julienrf** [comment]
@Bhudjo thanks for working on this! The solution consists in moving the definition of these operations to `mutable.SeqOps` instead of `mutable.IterableOps`. You might want to define `mapValuesInPlace` in `mutable.MapOps`, also.

**Comment by abuggin** [comment]
1. From Travis-CI/pr failure:
> The command "sbt -warn setupPublishCore generateBuildCharacterPropertiesFile publishLocal" exited with 1.

I can't reproduce this failure on my machine ☝️and I don't have a general idea of what problem this is about: `java.lang.NoClassDefFoundError: sbt/complete/DefaultParsers$` 😬

2. in jenkins build I have this error related to `Scala Presentation Compiler` (which I just discovered its existence)
```
[test] !!  9 - presentation/doc                          [output differs][duration 6.62s]
[test] % diff /home/jenkins/workspace/scala-2.13.x-validate-main/test/files/presentation/doc.check /home/jenkins/workspace/scala-2.13.x-validate-main/test/files/presentation/doc-presentation.log
[test] @@ -1 +1,2 @@
[test]  reload: Base.scala, Class.scala, Derived.scala
[test] +Unexpected foo method comment:None
```
From what I understood, the generated `doc-presentation.log` it's not supposed to have that additional `Unexpected foo method comment:None` line. 

Where that got created: no idea yet.

3. 
```
##### Log file '/home/jenkins/workspace/scala-2.13.x-validate-main/test/files/run/colltest1-run.log' from failed test #####

colltest1.scala:203: error: value retain is not a member of scala.collection.mutable.Map[String,String]
    val m2 = m1 retain ((k, v) => k == "N")
                ^
one error found
```

I am aware I just removed `retain` from `Map`, but since `LinkedHashMap` is an ordered collection, now I have the doubt if that `retain` should be left as possible method. If so, from where should `LinkedHashMap` get it? Being in doubt, I preferred to leave it broken and get suggestion from just removing it.

File: src/library/scala/collection/mutable/Iterable.scala [hunk 1]
```diff
@@ -17,17 +17,6 @@ trait Iterable[A]
...
-    *
-    *  @param f      the function to apply to each element.
-    *  @return       this $coll modified by replacing all elements with the
-    *                result of applying the given function `f` to each element
-    *                of this $coll.
-    */
-  def mapInPlace(f: A => A): this.type
-
-  @deprecated("Use `mapInPlace` instead", "2.13.0")
-  @`inline`final def transform(f: A => A): this.type = mapInPlace(f)
```
Comment (julienrf) [inline src/library/scala/collection/mutable/Iterable.scala:30]: @lrytz Why did we have this `transform` operation on `mutable.Iterable`? It does not seem to exist in 2.12.

File: test/files/run/colltest1.scala [hunk 2]
```diff
@@ -235,10 +235,8 @@ object Test extends App {
   setTest(immutable.HashSet())
 
   mapTest(Map())
-  mutableMapTest(mutable.Map())
   mapTest(immutable.Map())
   mapTest(immutable.TreeMap())
-  mutableMapTest(mutable.HashMap())
```
Comment (julienrf) [inline test/files/run/colltest1.scala:241]: Why can’t we keep these tests?

File: test/files/run/t7269.scala [hunk 3]
```diff
@@ -1,31 +0,0 @@
...
-    assert(11 == mapS.size)
-
-    // ConcurrentModificationException thrown in the following line
-    mapS.filterInPlace((i) => i % 2 == 0)
-    assert(6 == mapS.size)
-  }
-
-  testSet()
-  testMap()
-}
```
Comment (julienrf) [inline test/files/run/t7269.scala:31]: You should keep this test.

**Comment by julienrf** [comment]
3. Yes, I think we should keep `retain` in `mutable.Map` and `mutable.Set`. But at the same time, that means that having `filterInPlace` was not a so bad idea… What do you think, @Ichoran?

File: src/library/scala/collection/mutable/Iterable.scala (hunk 1 above)
Comment (lrytz) [reply src/library/scala/collection/mutable/Iterable.scala:30]: https://github.com/scala/scala/blob/v2.12.6/src/library/scala/collection/mutable/SeqLike.scala#L43 -- maybe the method should move to `mutable.SeqOps`? It's used in some tests (t4813.scala). The deprecated version here was added in https://github.com/scala/collection-strawman/pull/452.

Comment (julienrf) [reply src/library/scala/collection/mutable/Iterable.scala:30]: I agree with moving to `mutable.SeqOps`. (And that’s what’s done in this PR)

**Comment by julienrf** [comment]
Hey @Bhudjo, I have updated your work to include `mapValuesInPlace`. I’ve also squashed everything and rebased on the HEAD. Since I can not force push to your fork I pushed to mine and created another PR: #6910 (you are still the author of the commits). I suggest that we close this one and continue the work there.

**Comment by abuggin** [comment]
Thanks @julienrf 👍 
I did raise this PR open to collaborator, but yeah, I guess force push is not available.
Happy to continue breaking stuff on your PR 😹

**Pull request closed without merging**